enterprise/reporting/reporting.go

```diff
@@ -589,6 +589,10 @@ func (r *DefaultReporter) Report(ctx context.Context, metrics []*types.PUReporte
 			metric = transformMetricForPII(metric, getPIIColumnsToExclude())
 		}
 
+		if len(metric.StatusDetail.EventName) > config.GetInt("Reporting.eventNameMaxLength", 100) {
```

Comment:
why don't we trim down event names instead to the first 100 characters?

Reply:
We are not sending first 100 characters, we are sending `INVALID_EVENT_NAME`

Reply:
https://github.com/rudderlabs/rudder-server/pull/4244#discussion_r1429901263

Reply:
Cardinality can be a serious issue even with event names having less than 100 characters though, e.g. a uuid is 36 characters and ksuid is 28 characters.
Why are we coupling it with the length of characters?

Reply:
Initial thought is to trim them but when we looked at the data, 99.55% of eventNames longer than 100 characters are stringified JSONs or URLs. And rest of them can also be avoided with good instrumentation. So, we decided to go with this as this will also help with cardinality.